src/main/java/com/facebook/openwifirrm/modules/ProvMonitor.java

```diff
@@ -160,8 +194,79 @@ protected void syncDataToProv(
 			topo.values().stream().mapToInt(x -> x.size()).sum()
 		);
 
+		// Sync zone configs
+		deviceDataManager.updateZoneConfig(
+			configMap -> {
+				for (
+					Map.Entry<String, RRMDetails> entry : rrmDetails.entrySet()
+				) {
+					String venue = entry.getKey();
+					if (venue.isEmpty()) {
+						logger.error("Venue is blank for an RRM enabled zone");
+						continue;
+					}
+
+					String zone = venueIdToName.get(venue);
+					if (zone == null) {
+						logger.error(
+							"Venue name {} is not found in id mapping",
+							venue
+						);
+						continue;
+					}
+
+					DeviceConfig cfg = configMap
+						.computeIfAbsent(zone, k -> new DeviceConfig());
+					cfg.enableRRM =
+						cfg.enableConfig = cfg.enableWifiScan = true;
+
+					// read the details from the config
+					RRMDetails details = entry.getValue();
+					if (details == null) {
+						logger.error(
+							"No RRM details available for zone {} even though it has RRM enabled",
+							zone
+						);
+						continue;
+					}
+
+					cfg.schedule = new RRMSchedule();
+					if (details.rrm != null) {
```

Comment:
Both of these checks should be done before modifying `configMap` (don't add non-null but empty `cfg.schedule` -- null is OK)